cmd/doracled/cmd/gen_oracle_key.go

```diff
@@ -74,9 +77,33 @@ So please be cautious in using this command.`,
 				return err
 			}
 
+			// If FlagIsGenesisOracle is true, create a QueryClient
+			isGenesisOracle, err := cmd.Flags().GetBool(flags.FlagIsGenesisOracle)
+			if err != nil {
+				return err
+			}
+			if isGenesisOracle {
+				// get trusted block information
+				trustedBlockInfo, err := getTrustedBlockInfo(cmd)
+				if err != nil {
+					return fmt.Errorf("failed to get trusted block info: %w", err)
+				}
+
+				// initialize query client using trustedBlockInfo
+				queryClient, err := panacea.NewQueryClient(context.Background(), conf, *trustedBlockInfo)
+				if err != nil {
+					return fmt.Errorf("failed to initialize QueryClient: %w", err)
+				}
+				defer queryClient.Close()
+			}
+
 			return nil
 		},
 	}
+	cmd.Flags().Bool(flags.FlagIsGenesisOracle, false, "Check if it is the genesis oracle creation")
```

Comment:
I think this value is not needed.

This is because the `gen-oracle-key` command is executed only once by the first oracle.

In other words, `flags.FlagTrustedBlockHeight` and flags.FlagTrustedBlockHash` would be required values.

Reply:
I'm a little confused. Aren't all oracles running `gen-oracle-key` to generate `nodePrivKey`?

Reply:
No. Generate `OraclePrivKey`. :)

Reply:
Ah, sorry. I confused. Okay, I will fix it 👍